ravenframework/Simulation.py

```diff
@@ -688,78 +709,106 @@ def __readRunInfo(self,xmlNode,runInfoSkip,xmlFilename):
     """
-      This method is aimed to assemble and initialize
-      the step to make it ready to be executed
+      This method assembles and initializes the step to make it ready to be executed
       @ In, stepName, str, the step to initialize
       @ Out, (stepInputDict, stepInstance), tuple, tuple of step input dictionary and step instance
     """
     stepInstance = self.stepsDict[stepName]   # retrieve the instance of the step
+    if self.ranPreviously:
+      stepInstance.flushStep()
     self.raiseAMessage(f'-- Beginning {stepInstance.type} step "{stepName}" ... --')
     self.runInfoDict['stepName'] = stepName   # provide the name of the step to runInfoDict
     stepInputDict = {}                        # initialize the input dictionary for a step. Never use an old one!!!!!
     stepInputDict['Input' ] = []              # set the Input to an empty list
     stepInputDict['Output'] = []              # set the Output to an empty list
-    #fill the take a a step input dictionary just to recall: key= role played in the step b= Class, c= Type, d= user given name
+    # fill the take a step input dictionary just to recall: key= role played in the step b= Class, c= Type, d= user given name
     for role, entity, _, name in stepInstance.parList:
-      #Only for input and output we allow more than one object passed to the step, so for those we build a list
-      if role == 'Input' or role == 'Output':
+      # Only for input and output we allow more than one object passed to the step, so for those we build a list
+      if role == 'Input':
         stepInputDict[role].append(self.getEntity(entity, name))
+      elif role == 'Output':
+        if self.ranPreviously and entity == 'DataObjects':
+          # if simulation was run previously, output DataObjects need to be flushed
+          flushDataObject = self.getEntity(entity, name)
+          flushDataObject.flushDataObject()
+          # now add to stepInputDict
+          stepInputDict[role].append(flushDataObject)
+        else:
+          stepInputDict[role].append(self.getEntity(entity, name))
+      elif role == 'SolutionExport' and entity == 'DataObjects' and self.ranPreviously:
+        # if simulation was run previously, SolutionExport DataObjects need to be flushed
+        flushDataObject = self.getEntity(entity, name)
+        flushDataObject.flushDataObject()
+        # now add to stepInputDict
+        stepInputDict[role] = flushDataObject
+      elif role == 'Optimizer' and self.ranPreviously:
+        # if simulation was run previously, Optimizer needs to be flushed
+        flushOptimizer = self.getEntity(entity, name)
+        flushOptimizer.flushOptimizer()
+        # now add to stepInputDict
+        stepInputDict[role] = flushOptimizer
+      elif role == 'Sampler' and self.ranPreviously:
+        # if simulation was run previously, Sampler needs to be flushed
+        flushSampler = self.getEntity(entity, name)
+        flushSampler.flushSampler()
```

Comment:
wouldn't a common "flush" method name allow us to reduce the code added here? Is there a reason to preserve specific flush method names?

Reply:
Hm, I'll have to look into this. I'm not sure how that would work with the relationship between Optimizers and Samplers.

Reply:
Okay, looks like a common "flush" method will work.